### PR TITLE
refactor(table): change the paddings in the expandable content

### LINKED ISSUE #1507: Update Expandable Table example: move content back to table column.
As a follow up to #1447... @kybaker would like the content in the expandable row to sit inside the column that lines up with the column in the row above. 

This can be produced without any changes to the code. The expandable row should have no padding, and use `<td>'s` and adjust `colspan`.

### PR DESCRIPTION
**Pull request by christiemolloy**
fix #1507

### CODE REVIEW

File: src/patternfly/components/Table/table.scss [hunk 1]
```diff
@@ -385,14 +383,22 @@
...
 
-  // no padding on standard td
-  td:only-child {
-    padding: 0;
-    padding-left: var(--pf-c-table__expandable-row--before--Width); // set padding-left to adjust for left border.
+  /* stylelint-disable */
+  td {
+    &.pf-m-no-padding {
+      padding: 0;
+      padding-left: var(--pf-c-table__expandable-row--before--Width); // set padding-left to adjust for left border.
```
Comment (mcoker) [inline src/patternfly/components/Table/table.scss]: Can you write this as `padding: 0 0 0 var(--pf-c-table__expandable-row--before--Width);`?

File: src/patternfly/components/Table/examples/table-compound-expansion-example.hbs [hunk 2]
```diff
@@ -26,19 +26,19 @@
       {{#> table-th table-th--data-label="Repository Name"}}
         <a href="#">siemur/test-space</a>
       {{/table-th}}
-      {{#> table-td table-td--compound-expansion-toggle="true" table-td--modifier="pf-m-expanded" table-td--data-label="Branches"}}
+      {{#> table-td table-td--compound-expansion-toggle="true" table-td--modifier="pf-m-expanded pf-m-no-padding" table-td--data-label="Branches"}}
```
Comment (mcoker) [inline src/patternfly/components/Table/examples/table-compound-expansion-example.hbs]: did you mean to add `pf-m-no-padding` to the compound expansion toggle cells? Doesn't look like they have any padding by default.

File: src/patternfly/components/Table/examples/table-compound-expansion-example.hbs [hunk 3]
```diff
@@ -57,22 +57,22 @@
     {{/table-tr}}
 
     {{#> table-tr table-tr--expandable="true" table-tr--IsExpanded="true"}}
-      {{#> table-td table-td--attribute='colspan="7"'}}
-        {{#> table-nested table--id=(concat table--id '-nested-table-1')}}
+      {{#> table-td table-td--attribute='colspan="7"' table-td--modifier="pf-m-no-padding"}}
+        {{#> table-nested table--id=(concat table--id '-nested-table-1"')}}
```
Comment (mcoker) [inline src/patternfly/components/Table/examples/table-compound-expansion-example.hbs]: need to remove the extra double quotes in all of these nested table `table--id`'s. those were fixed in https://github.com/patternfly/patternfly-next/pull/1614/files

```suggestion
        {{#> table-nested table--id=(concat table--id '-nested-table-1')}}
```

File: src/patternfly/components/Table/examples/table-expandable-example.hbs [hunk 4]
```diff
@@ -47,19 +47,22 @@
     {{/table-tr}}
 
     {{#> table-tr table-tr--expandable="true" table-tr--IsExpanded="true"}}
-      {{#> table-td table-td--attribute=(concat 'colspan="7" id="' table--id '-content1"')}}
+      {{#> table-td}}{{/table-td}}
+      {{#> table-td}}{{/table-td}}
+      {{#> table-td table-td--attribute=(concat 'colspan="4 id="' table--id '-content1"')}}
         <div class="pf-c-table__expandable-row-content">
```
Comment (mcoker) [inline src/patternfly/components/Table/examples/table-expandable-example.hbs]: should we create an hbs file for this? we use it a lot.

File: src/patternfly/components/Table/examples/table-expandable-example.hbs [hunk 5]
```diff
@@ -47,19 +47,22 @@
...
       {{/table-td}}
+      {{#> table-td}}{{/table-td}}
     {{/table-tr}}
   {{/table-tbody}}
 
-  {{#> table-tbody table-tbody--modifier="pf-m-expanded"}}
+  {{#> table-tbody}}
     {{#> table-tr}}
-      {{#> table-td table-td--toggle="true" table-td--button--attribute=(concat 'aria-labelledby="' table--id '-node2 expandable-toggle2" id="expandable-toggle2" aria-label="Details" aria-controls="' table--id '-content2"')}}{{/table-td}}
+      {{#> table-td table-td--toggle="true" table-td--button--attribute=(concat 'aria-labelledby="' table--id '-node1 expandable-toggle2" id="expandable-toggle2" aria-label="Details" aria-controls="' table--id '-content2"')}}{{/table-td}}
```
Comment (mcoker) [inline src/patternfly/components/Table/examples/table-expandable-example.hbs]:
```suggestion
      {{#> table-td table-td--toggle="true" table-td--button--attribute=(concat 'aria-labelledby="' table--id '-node2 expandable-toggle2" id="expandable-toggle2" aria-label="Details" aria-controls="' table--id '-content2"')}}{{/table-td}}
```

File: src/patternfly/components/Table/examples/table-compound-expansion-example.hbs (hunk 2 above)
Comment (christiemolloy) [reply src/patternfly/components/Table/examples/table-compound-expansion-example.hbs]: yes that was a mistake thanks for catching that!

File: src/patternfly/components/Table/examples/table-expandable-example.hbs (hunk 4 above)
Comment (christiemolloy) [reply src/patternfly/components/Table/examples/table-expandable-example.hbs]: yes created one

**Comment by mcoker** [comment]
Can you verify this is what it's supposed to look like on mobile? 
Screen Shot 2019-03-25 at 2 06 15 PM

**Comment by christiemolloy** [comment]
@mcoker checked designs, I think its meant to be 32px padding left/right for the expandable content. As before it was 32px padding-left/right for the main content, then 64px for the expandable content. But after the change to 16px padding-left/right, I assume it will still be double padding. @kybaker can you review this padding on desktop and mobile ?

**Comment by christiemolloy** [comment]
@mcoker updated spacing here after speaking with @kybaker

File: src/patternfly/components/Table/docs/table-expandable.md [hunk 6]
```diff
@@ -23,6 +23,7 @@ Note: To apply padding to `.pf-c-table__expandable-row`, wrap the content in `.p
 
 | Class | Applied To | Outcome |
 | -- | -- | -- |
-| `.pf-c-table__expandable-row`         | `<tr>`                                                                | Initiates an expandable row |
-| `.pf-c-table__expandable-row-content` | `.pf-c-table__expandable-row` > `<td>` > `<div>`                      | Initiates an expandable row content wrapper. |
-| `.pf-m-expanded`                      | `.pf-c-table__toggle` > `.pf-c-button`, `.pf-c-table__expandable-row` | Modifies for expanded state. |
+| `.pf-c-table__expandable-row` | `<tr>` | Initiates an expandable row. |
+| `.pf-c-table__expandable-row-content` | `.pf-c-table__expandable-row` > `<td>` > `<div>` | Initiates an expandable row content wrapper. |
+| `.pf-m-expanded` | `.pf-c-table__toggle` > `.pf-c-button`, `.pf-c-table__expandable-row` | Modifies for expanded state. |
+| `.pf-m-no-padding` | `.pf-c-expandable-row` > `<td>` | Modifies the expandable row to have no padding.
```
Comment (mcoker) [inline src/patternfly/components/Table/docs/table-expandable.md]:
```suggestion
| `.pf-m-no-padding` | `.pf-c-table__expandable-row` > `<td>` | Modifies the expandable row to have no padding.
```

**Pull request merged**